Reqnroll.VisualStudio/Editor/Services/UnusedStepDefinitionsFinder.cs

```diff
@@ -0,0 +1,127 @@
+﻿using Reqnroll.VisualStudio.ProjectSystem;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+#nullable disable
+namespace Reqnroll.VisualStudio.Editor.Services;
+
+public class UnusedStepDefinitionsFinder
+{
+    private readonly IIdeScope _ideScope;
+
+    public UnusedStepDefinitionsFinder(IIdeScope ideScope)
+    {
+        _ideScope = ideScope;
+    }
+
+    public IEnumerable<ProjectStepDefinitionBinding> FindUnused(ProjectStepDefinitionBinding[] stepDefinitions,
+                                                        string[] featureFiles, DeveroomConfiguration configuration)
+    {
+        return featureFiles.SelectMany(ff => FindUnused(stepDefinitions, ff, configuration));
+    }
+
+    public IEnumerable<ProjectStepDefinitionBinding> FindUnused(ProjectStepDefinitionBinding[] stepDefinitions, 
+        string featureFilePath, DeveroomConfiguration configuration) => 
+        LoadContent(featureFilePath, out string featureFileContent)
+        ? FindUsagesFromContent(stepDefinitions, featureFileContent, featureFilePath, configuration)
+        : Enumerable.Empty<ProjectStepDefinitionBinding>();
+
+    private IEnumerable<ProjectStepDefinitionBinding> FindUsagesFromContent(ProjectStepDefinitionBinding[] stepDefinitions, string featureFileContent, string featureFilePath, DeveroomConfiguration configuration)
+    {
+        var dialectProvider = ReqnrollGherkinDialectProvider.Get(configuration.DefaultFeatureLanguage);
+        var parser = new DeveroomGherkinParser(dialectProvider, _ideScope.MonitoringService);
+        parser.ParseAndCollectErrors(featureFileContent, _ideScope.Logger,
+            out var gherkinDocument, out _);
+
+        var featureNode = gherkinDocument?.Feature;
+        if (featureNode == null)
+            return Enumerable.Empty<ProjectStepDefinitionBinding>();
+
+        var dummyRegistry = ProjectBindingRegistry.FromBindings(stepDefinitions);
```

Comment:
Instead of this you should get an instance of the `IDiscoveryService` from the project scope (like in `DefineStepsCommand` [here](https://github.com/reqnroll/Reqnroll.VisualStudio/blob/main/Reqnroll.VisualStudio/Editor/Commands/DefineStepsCommand.cs#L154)) in the command and pass it to the finder class.

In the finder class, you can get the binding registry with

```
var bindingRegistry = _discoveryService.BindingRegistryCache.Value;
```

this way you will not need the `stepDefinitions` param all the way up.